test/compiler/irpasses.jl

```diff
@@ -758,3 +758,20 @@ let # effect-freeness computation for array allocation
         nothing
     end
 end
+
+# allow branch folding to look at type information
+let
+    function fcond()
+        cond = 1 + 1 == 2
+        if !cond
+            gcd(24, 36)
+        else
+            gcd(64, 128)
+        end
+    end
+    ci = (@code_typed optimize=:false fcond()) |> first
```

Comment:
`@code_typed` needs `using InteractiveUtils`. `code_typed` is available without it though.

Reply:
Oops, fixed